### PR TITLE
feat: implement dataDeal query client

### LINKED ISSUE #537: Add CLI: DataDeal Query
### Implement
* GetDeals
* GetDeal
* GetCertificates
* GetCertificate

### PR DESCRIPTION
**Pull request by gyuguen**
close #537

### CODE REVIEW

File: x/datadeal/keeper/grpc_query_deal.go [hunk 1]
```diff
@@ -24,17 +26,79 @@ func (k Keeper) Deal(goCtx context.Context, req *types.QueryDealRequest) (*types
...
-	panic("implement me")
+func (k Keeper) Certificates(goCtx context.Context, req *types.QueryCertificates) (*types.QueryCertificatesResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	store := ctx.KVStore(k.storeKey)
+	certsStore := prefix.NewStore(store, types.CertificateKey)
```
Comment (H4NLee) [inline x/datadeal/keeper/grpc_query_deal.go:68]: The deal ID probably should be included in prefix.

File: x/datadeal/keeper/grpc_query_deal.go [hunk 2]
```diff
@@ -24,17 +26,79 @@ func (k Keeper) Deal(goCtx context.Context, req *types.QueryDealRequest) (*types
...
+		if err != nil {
+			return err
+		}
+		certs = append(certs, &cert)
+		return nil
+	})
+
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
```
Comment (H4NLee) [inline x/datadeal/keeper/grpc_query_deal.go:83]:
```suggestion
	if err != nil {
		return nil, status.Error(codes.Internal, err.Error())
	}
```

**Review by inchori — APPROVED** [review]
LGTM

**Review by youngjoon-lee — APPROVED** [review]
lgtm

**Review by audtlr24 — APPROVED** [review]
lgtm

File: x/datadeal/keeper/grpc_query_deal.go (hunk 2 above)
Comment (gyuguen) [reply x/datadeal/keeper/grpc_query_deal.go:83]: Umm I don't know what's different :)

Comment (H4NLee) [reply x/datadeal/keeper/grpc_query_deal.go:83]: just removed empty line between error occurrence and check.

Comment (gyuguen) [reply x/datadeal/keeper/grpc_query_deal.go:83]: Aha I got it.

File: x/datadeal/keeper/grpc_query_deal.go (hunk 1 above)
Comment (gyuguen) [reply x/datadeal/keeper/grpc_query_deal.go]: Thanks. fixed it(https://github.com/medibloc/panacea-core/pull/539/commits/0818bb2f49fb664dc8d8434ed545e68157e90b6a)

**Review by H4NLee — APPROVED** [review]
lgtm

**Pull request merged**